src/source.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2013-2015, The Rust-GNOME Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+//! GSourceFunc functions
+
+use std::cell::RefCell;
+use std::ops::DerefMut;
+use std::mem::transmute;
+use ffi::{gboolean, gpointer, g_idle_add_full, g_timeout_add_full, g_timeout_add_seconds_full};
+use translate::ToGlib;
+
+pub struct Continue(pub bool);
+
+impl ToGlib for Continue {
+    type GlibType = gboolean;
+
+    #[inline]
+    fn to_glib(&self) -> gboolean {
+        self.0.to_glib()
+    }
+}
+
+
+// Box::into_raw stability workaround
+unsafe fn into_raw<T>(b: Box<T>) -> *mut T { transmute(b) }
+
+extern "C" fn trampoline(func: &RefCell<Box<FnMut() -> Continue + 'static>>) -> gboolean {
+    func.borrow_mut().deref_mut()().to_glib()
+}
+
+extern "C" fn destroy_closure(ptr: gpointer) {
+    unsafe {
+        // Box::from_raw API stability workaround
+        let ptr = ptr as *mut RefCell<Box<FnMut() -> Continue + 'static>>;
+        let _: Box<RefCell<Box<FnMut() -> Continue + 'static>>> = transmute(ptr);
+    }
+}
+
+const G_PRIORITY_DEFAULT: i32 = 0;
+const G_PRIORITY_DEFAULT_IDLE: i32 = 200;
+
+
+/// Adds a function to be called whenever there are no higher priority events pending to the default main loop.
+/// The function is given the default idle priority, G_PRIORITY_DEFAULT_IDLE.
+/// If the function returns FALSE it is automatically removed from
```

Comment:
This will mislead the users because we ask them to return `Continue` not `bool`. The docs likely need some massaging to better fit changed names and signatures. Maybe you can add an example.

And I believe the Rust style requires identifiers and expressions to be quoted with backticks.